R/release.R

```diff
@@ -361,29 +363,34 @@ check_github_has_SHA <- function(SHA = gert::git_info(repo = git_repo())$commit,
 }
 
 get_release_news <- function(SHA = gert::git_info(repo = git_repo())$commit,
-                             tr = target_repo(github_get = TRUE)) {
+                             tr = target_repo(github_get = TRUE),
+                             on_cran = !is.null(cran_version())) {
   HEAD <- gert::git_info(repo = git_repo())$commit
 
   if (HEAD == SHA) {
     news_path <- proj_path("NEWS.md")
+    news <- if (file_exists(news_path)) read_utf8(news_path) else NULL
   } else {
-    news_path <- withr::local_tempfile()
-    gh <- purrr::possibly(gh_tr(tr), otherwise = NULL)
-    gh(
-      "/repos/{owner}/{repo}/contents/{path}",
-      path = "NEWS.md", ref = SHA,
-      .destfile = news_path,
-      .accept = "application/vnd.github.v3.raw"
+    news <- tryCatch(
+      read_github_file(
+        tr$repo_spec,
+        path = "NEWS.md",
+        ref = SHA,
+        host = tr$api_url
+      ),
+      github_error = NULL
     )
   }
 
-  if (file_exists(news_path)) {
-    news <- news_latest(read_utf8(news_path))
+  if (is.null(news)) {
+    ui_oops("
+      Can't find {ui_path('NEWS.md')} in the released package source.
+      usethis consults this file for release notes.
+      Call {ui_code('usethis::use_news_md()')} to set this up for the future.")
+    if (on_cran) "-- no release notes --" else "Initial release"
```

Comment:
This feels like it will cause a productive amount of annoyance for anyone who sees it. Their current release will go through, but they're encouraged to create `NEWS.md`. And the release notes can be edited if they're really unhappy with the text I've made up.